readthedocs/oauth/services/githubapp.py

```diff
@@ -0,0 +1,473 @@
+        remote_repo.default_branch = gh_repo.default_branch
+        remote_repo.clone_url = gh_repo.clone_url
+
+        # NOTE: Only one installation of our APP should give access to a repository.
+        # This should only happen if our data is out of sync.
+        if (
+            remote_repo.github_app_installation
+            and remote_repo.github_app_installation != self.installation
+        ):
+            log.info(
+                "Repository linked to another installation",
+                repository_id=remote_repo.remote_id,
+                old_installation_id=remote_repo.github_app_installation.installation_id,
+                new_installation_id=self.installation.installation_id,
+            )
+        remote_repo.github_app_installation = self.installation
+
+        remote_repo.organization = None
+        if gh_repo.owner.type == GitHubAccountType.ORGANIZATION:
+            # NOTE: The owner object doesn't have all attributes of an organization,
+            # so we need to fetch the organization object.
+            remote_repo.organization = self.update_or_create_organization(gh_repo.owner.login)
+
+        remote_repo.save()
+        self._resync_collaborators(gh_repo, remote_repo)
+        return remote_repo
+
+    # NOTE: normally, this should cache only one organization at a time, but just in case...
+    @lru_cache(maxsize=50)
+    def _get_gh_organization(self, login: str) -> GHOrganization:
+        """Get a GitHub organization object given its login identifier."""
+        return self.installation_client.get_organization(login)
+
+    # NOTE: normally, this should cache only one organization at a time, but just in case...
+    @lru_cache(maxsize=50)
+    def update_or_create_organization(self, login: str) -> RemoteOrganization:
+        """
+        Create or update a remote organization from its login identifier.
+
+        We also sync the members of the organization with the database.
+        This doesn't sync the repositories of the organization,
+        since the installation is the one that lists the repositories it has access to.
+
+        This method is cached, since we need to update the organization only once per sync of an installation.
+        """
+        gh_org = self._get_gh_organization(login)
+        remote_org, _ = RemoteOrganization.objects.get_or_create(
+            remote_id=str(gh_org.id),
+            vcs_provider=self.vcs_provider_slug,
+        )
+        remote_org.slug = gh_org.login
+        remote_org.name = gh_org.name
+        # NOTE: do we need the email of the organization?
+        remote_org.email = gh_org.email
+        remote_org.avatar_url = gh_org.avatar_url or self.default_org_avatar_url
+        remote_org.url = gh_org.html_url
+        remote_org.save()
+        self._resync_organization_members(gh_org, remote_org)
+        return remote_org
+
+    def _resync_collaborators(self, gh_repo: GHRepository, remote_repo: RemoteRepository):
```

Comment:
What are collaborators? Is this permissions?

Reply:
Basically anyone with access to the repository.

https://docs.github.com/en/rest/collaborators/collaborators?apiVersion=2022-11-28#list-repository-collaborators

Reply:
Added that link to the docstring